File: graphics/examplesCxx/quadricCut.cxx
#include "vtkBooleanTexture.h"
#include "vtkTexture.h"
#include "vtkActor.h"
#include "vtkRenderer.h"
#include "vtkRenderWindow.h"
#include "vtkRenderWindowInteractor.h"
#include "vtkQuadric.h"
#include "vtkSphereSource.h"
#include "vtkImplicitTextureCoords.h"
#include "vtkDataSetMapper.h"
#include "vtkActor.h"

float positions[][3] = {
	{-4,4,0}, {-2,4,0}, {0,4,0}, {2,4,0},
	{-4,2,0}, {-2,2,0}, {0,2,0}, {2,2,0},
	{-4,0,0}, {-2,0,0}, {0,0,0}, {2,0,0},
	{-4,-2,0}, {-2,-2,0}, {0,-2,0}, {2,-2,0}
};

vtkBooleanTexture *makeBooleanTexture (int, int, int);

#include "SaveImage.h"

int main( int argc, char *argv[] )
{
  int i;
  vtkBooleanTexture *aBoolean[16];
  vtkTexture *aTexture[16];
  vtkActor *anActor[16];

  vtkRenderer *aren = vtkRenderer::New();
  vtkRenderWindow *renWin = vtkRenderWindow::New();
    renWin->AddRenderer(aren);
  vtkRenderWindowInteractor *iren = vtkRenderWindowInteractor::New();
    iren->SetRenderWindow(renWin);

  // define two elliptical cylinders
  vtkQuadric *quadric1 = vtkQuadric::New();
      quadric1->SetCoefficients (1,2,0,0,0,0,0,0,0,-.07);

  vtkQuadric *quadric2 = vtkQuadric::New();
      quadric2->SetCoefficients (2,1,0,0,0,0,0,0,0,-.07);

  // create a sphere for all to use
  vtkSphereSource *aSphere = vtkSphereSource::New();
    aSphere->SetPhiResolution(100); aSphere->SetThetaResolution(100); 

  // create texture coordianates for all
  vtkImplicitTextureCoords *tcoords = vtkImplicitTextureCoords::New();
      tcoords->SetInput(aSphere->GetOutput());
      tcoords->SetRFunction(quadric1);
      tcoords->SetSFunction(quadric2);

  vtkDataSetMapper *aMapper = vtkDataSetMapper::New();
      aMapper->SetInput (tcoords->GetOutput ());

  // create a mapper, sphere and texture map for each case
  for (i = 0; i < 16; i++) 
    {
    aBoolean[i] = makeBooleanTexture (i, 256, 1);

    aTexture[i] = vtkTexture::New();
      aTexture[i]->SetInput (aBoolean[i]->GetOutput());
      aTexture[i]->InterpolateOff();
      aTexture[i]->RepeatOff ();
    anActor[i] = vtkActor::New();
      anActor[i]->SetMapper (aMapper);
      anActor[i]->SetTexture (aTexture[i]);
      anActor[i]->SetPosition ( &positions[i][0]);
      anActor[i]->SetScale (2.0, 2.0, 2.0);
    aren->AddActor (anActor[i]);
    }

  aren->SetBackground (0.4392,0.5020,0.5647);
  aren->GetActiveCamera()->Zoom(1.4);
  renWin->DoubleBufferOff();
  renWin->SetSize(300,300);

  // interact with data
  //renWin->SetFileName("plate42.ppm");
  renWin->Render();
  //renWin->SaveImageAsPPM();

  SAVEIMAGE( renWin );
  
  iren->Start();

  // Clean up
  aren->Delete();
  renWin->Delete();
  iren->Delete();
  quadric1->Delete();
  quadric2->Delete();
  aSphere->Delete();
  tcoords->Delete();
  aMapper->Delete();
  for (i=0; i<16; i++)
    {
    aBoolean[i]->Delete();
    aTexture[i]->Delete();
    anActor[i]->Delete();
    }
}

static unsigned char solid[] = {255, 255};
static unsigned char clear[] = {255, 0};
static unsigned char edge[] =  {0, 255};

vtkBooleanTexture *makeBooleanTexture (int caseNumber, int resolution, int thickness)
{
  vtkBooleanTexture *booleanTexture = vtkBooleanTexture::New();

  booleanTexture->SetXSize (resolution);
  booleanTexture->SetYSize (resolution);
  booleanTexture->SetThickness (thickness);

  switch (caseNumber) 
    {
    case 0:
      booleanTexture->SetInIn (solid);
      booleanTexture->SetOutIn (solid);
      booleanTexture->SetInOut (solid);
      booleanTexture->SetOutOut (solid);
      booleanTexture->SetOnOn (solid);
      booleanTexture->SetOnIn (solid);
      booleanTexture->SetOnOut (solid);
      booleanTexture->SetInOn (solid);
      booleanTexture->SetOutOn (solid);
      break;
    case 1:	/* cut inside 1 */
      booleanTexture->SetInIn (clear);
      booleanTexture->SetOutIn (solid);
      booleanTexture->SetInOut (solid);
      booleanTexture->SetOutOut (solid);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (edge);
      booleanTexture->SetOnOut (solid);
      booleanTexture->SetInOn (edge);
      booleanTexture->SetOutOn (solid);
      break;
    case 2:	/* cut outside 1 */
      booleanTexture->SetInIn (solid);
      booleanTexture->SetOutIn (clear);
      booleanTexture->SetInOut (solid);
      booleanTexture->SetOutOut (solid);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (edge);
      booleanTexture->SetOnOut (solid);
      booleanTexture->SetInOn (solid);
      booleanTexture->SetOutOn (edge);
      break;
    case 3:	/* cut all 1 */
      booleanTexture->SetInIn (clear);
      booleanTexture->SetOutIn (clear);
      booleanTexture->SetInOut (solid);
      booleanTexture->SetOutOut (solid);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (clear);
      booleanTexture->SetOnOut (solid);
      booleanTexture->SetInOn (edge);
      booleanTexture->SetOutOn (edge);
      break;
    case 4:
      booleanTexture->SetInIn (solid);
      booleanTexture->SetOutIn (solid);
      booleanTexture->SetInOut (clear);
      booleanTexture->SetOutOut (solid);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (solid);
      booleanTexture->SetOnOut (edge);
      booleanTexture->SetInOn (edge);
      booleanTexture->SetOutOn (solid);
      break;
    case 5:
      booleanTexture->SetInIn (clear);
      booleanTexture->SetOutIn (solid);
      booleanTexture->SetInOut (clear);
      booleanTexture->SetOutOut (solid);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (edge);
      booleanTexture->SetOnOut (edge);
      booleanTexture->SetInOn (clear);
      booleanTexture->SetOutOn (solid);
      break;
    case 6:
      booleanTexture->SetInIn (solid);
      booleanTexture->SetOutIn (clear);
      booleanTexture->SetInOut (clear);
      booleanTexture->SetOutOut (solid);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (edge);
      booleanTexture->SetOnOut (edge);
      booleanTexture->SetInOn (edge);
      booleanTexture->SetOutOn (edge);
      break;
    case 7:
      booleanTexture->SetInIn (clear);
      booleanTexture->SetOutIn (clear);
      booleanTexture->SetInOut (clear);
      booleanTexture->SetOutOut (solid);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (clear);
      booleanTexture->SetOnOut (edge);
      booleanTexture->SetInOn (clear);
      booleanTexture->SetOutOn (edge);
      break;
    case 8:
      booleanTexture->SetInIn (solid);
      booleanTexture->SetOutIn (solid);
      booleanTexture->SetInOut (solid);
      booleanTexture->SetOutOut (clear);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (solid);
      booleanTexture->SetOnOut (edge);
      booleanTexture->SetInOn (solid);
      booleanTexture->SetOutOn (edge);
      break;
    case 9:
      booleanTexture->SetInIn (clear);
      booleanTexture->SetInOut (solid);
      booleanTexture->SetOutIn (solid);
      booleanTexture->SetOutOut (clear);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (edge);
      booleanTexture->SetOnOut (edge);
      booleanTexture->SetInOn (edge);
      booleanTexture->SetOutOn (edge);
      break;
    case 10:
      booleanTexture->SetInIn (solid);
      booleanTexture->SetInOut (solid);
      booleanTexture->SetOutIn (clear);
      booleanTexture->SetOutOut (clear);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (edge);
      booleanTexture->SetOnOut (edge);
      booleanTexture->SetInOn (solid);
      booleanTexture->SetOutOn (clear);
      break;
    case 11:
      booleanTexture->SetInIn (clear);
      booleanTexture->SetInOut (solid);
      booleanTexture->SetOutIn (clear);
      booleanTexture->SetOutOut (clear);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (clear);
      booleanTexture->SetOnOut (edge);
      booleanTexture->SetInOn (edge);
      booleanTexture->SetOutOn (clear);
      break;
    case 12:
      booleanTexture->SetInIn (solid);
      booleanTexture->SetInOut (clear);
      booleanTexture->SetOutIn (solid);
      booleanTexture->SetOutOut (clear);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (solid);
      booleanTexture->SetOnOut (clear);
      booleanTexture->SetInOn (edge);
      booleanTexture->SetOutOn (edge);
      break;
    case 13:
      booleanTexture->SetInIn (clear);
      booleanTexture->SetInOut (clear);
      booleanTexture->SetOutIn (solid);
      booleanTexture->SetOutOut (clear);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (edge);
      booleanTexture->SetOnOut (clear);
      booleanTexture->SetInOn (clear);
      booleanTexture->SetOutOn (edge);
      break;
    case 14:
      booleanTexture->SetInIn (solid);
      booleanTexture->SetInOut (clear);
      booleanTexture->SetOutIn (clear);
      booleanTexture->SetOutOut (clear);
      booleanTexture->SetOnOn (edge);
      booleanTexture->SetOnIn (edge);
      booleanTexture->SetOnOut (clear);
      booleanTexture->SetInOn (edge);
      booleanTexture->SetOutOn (clear);
      break;
    case 15:
      booleanTexture->SetInIn (clear);
      booleanTexture->SetInOut (clear);
      booleanTexture->SetOutIn (clear);
      booleanTexture->SetOutOut (clear);
      booleanTexture->SetOnOn (clear);
      booleanTexture->SetOnIn (clear);
      booleanTexture->SetOnOut (clear);
      booleanTexture->SetInOn (clear);
      booleanTexture->SetOutOn (clear);
      break;
    }

  return booleanTexture;
}

File: Common/DataModel/Testing/Python/quadricCut.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-



from vtkmodules.vtkCommonDataModel import vtkQuadric
from vtkmodules.vtkFiltersSources import vtkSphereSource
from vtkmodules.vtkFiltersTexture import vtkImplicitTextureCoords
from vtkmodules.vtkImagingHybrid import vtkBooleanTexture
from vtkmodules.vtkRenderingCore import (
    vtkActor,
    vtkDataSetMapper,
    vtkRenderWindow,
    vtkRenderWindowInteractor,
    vtkRenderer,
    vtkTexture,
)
import vtkmodules.vtkInteractionStyle
import vtkmodules.vtkRenderingFreeType
import vtkmodules.vtkRenderingOpenGL2
import vtkmodules.test.Testing
from vtkmodules.util.misc import vtkGetDataRoot
VTK_DATA_ROOT = vtkGetDataRoot()

class quadricCut(vtkmodules.test.Testing.vtkTest):

    def testQuadricCut(self):

        solidTexture = (255, 255)
        clearTexture = (255, 0)
        edgeTexture = (0, 255)

        def makeBooleanTexture(caseNumber, resolution, thickness):
            #global solidTexture, clearTexture, edgeTexture
            booleanTexturecaseNumber = vtkBooleanTexture()

            booleanTexturecaseNumber.SetXSize(resolution)
            booleanTexturecaseNumber.SetYSize(resolution)
            booleanTexturecaseNumber.SetThickness(thickness)

            if caseNumber == 0:
                booleanTexturecaseNumber.SetInIn(solidTexture)
                booleanTexturecaseNumber.SetOutIn(solidTexture)
                booleanTexturecaseNumber.SetInOut(solidTexture)
                booleanTexturecaseNumber.SetOutOut(solidTexture)
                booleanTexturecaseNumber.SetOnOn(solidTexture)
                booleanTexturecaseNumber.SetOnIn(solidTexture)
                booleanTexturecaseNumber.SetOnOut(solidTexture)
                booleanTexturecaseNumber.SetInOn(solidTexture)
                booleanTexturecaseNumber.SetOutOn(solidTexture)
            elif caseNumber == 1:
                booleanTexturecaseNumber.SetInIn(clearTexture)
                booleanTexturecaseNumber.SetOutIn(solidTexture)
                booleanTexturecaseNumber.SetInOut(solidTexture)
                booleanTexturecaseNumber.SetOutOut(solidTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(edgeTexture)
                booleanTexturecaseNumber.SetOnOut(solidTexture)
                booleanTexturecaseNumber.SetInOn(edgeTexture)
            elif caseNumber == 2:
                booleanTexturecaseNumber.SetInIn(solidTexture)
                booleanTexturecaseNumber.SetOutIn(clearTexture)
                booleanTexturecaseNumber.SetInOut(solidTexture)
                booleanTexturecaseNumber.SetOutOut(solidTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(edgeTexture)
                booleanTexturecaseNumber.SetOnOut(solidTexture)
                booleanTexturecaseNumber.SetInOn(solidTexture)
                booleanTexturecaseNumber.SetOutOn(edgeTexture)
            elif caseNumber == 3:
                booleanTexturecaseNumber.SetInIn(clearTexture)
                booleanTexturecaseNumber.SetOutIn(clearTexture)
                booleanTexturecaseNumber.SetInOut(solidTexture)
                booleanTexturecaseNumber.SetOutOut(solidTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(clearTexture)
                booleanTexturecaseNumber.SetOnOut(solidTexture)
                booleanTexturecaseNumber.SetInOn(edgeTexture)
                booleanTexturecaseNumber.SetOutOn(edgeTexture)
            elif caseNumber == 4:
                booleanTexturecaseNumber.SetInIn(solidTexture)
                booleanTexturecaseNumber.SetOutIn(solidTexture)
                booleanTexturecaseNumber.SetInOut(clearTexture)
                booleanTexturecaseNumber.SetOutOut(solidTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(solidTexture)
                booleanTexturecaseNumber.SetOnOut(edgeTexture)
                booleanTexturecaseNumber.SetInOn(edgeTexture)
                booleanTexturecaseNumber.SetOutOn(solidTexture)
            elif caseNumber == 5:
                booleanTexturecaseNumber.SetInIn(clearTexture)
                booleanTexturecaseNumber.SetOutIn(solidTexture)
                booleanTexturecaseNumber.SetInOut(clearTexture)
                booleanTexturecaseNumber.SetOutOut(solidTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(edgeTexture)
                booleanTexturecaseNumber.SetOnOut(edgeTexture)
                booleanTexturecaseNumber.SetInOn(clearTexture)
                booleanTexturecaseNumber.SetOutOn(solidTexture)
            elif caseNumber == 6:
                booleanTexturecaseNumber.SetInIn(solidTexture)
                booleanTexturecaseNumber.SetOutIn(clearTexture)
                booleanTexturecaseNumber.SetInOut(clearTexture)
                booleanTexturecaseNumber.SetOutOut(solidTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(edgeTexture)
                booleanTexturecaseNumber.SetOnOut(edgeTexture)
                booleanTexturecaseNumber.SetInOn(edgeTexture)
                booleanTexturecaseNumber.SetOutOn(edgeTexture)
            elif caseNumber == 7:
                booleanTexturecaseNumber.SetInIn(clearTexture)
                booleanTexturecaseNumber.SetOutIn(clearTexture)
                booleanTexturecaseNumber.SetInOut(clearTexture)
                booleanTexturecaseNumber.SetOutOut(solidTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(clearTexture)
                booleanTexturecaseNumber.SetOnOut(edgeTexture)
                booleanTexturecaseNumber.SetInOn(clearTexture)
                booleanTexturecaseNumber.SetOutOn(edgeTexture)
            elif caseNumber == 8:
                booleanTexturecaseNumber.SetInIn(solidTexture)
                booleanTexturecaseNumber.SetOutIn(solidTexture)
                booleanTexturecaseNumber.SetInOut(solidTexture)
                booleanTexturecaseNumber.SetOutOut(clearTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(solidTexture)
                booleanTexturecaseNumber.SetOnOut(edgeTexture)
                booleanTexturecaseNumber.SetInOn(solidTexture)
                booleanTexturecaseNumber.SetOutOn(edgeTexture)
            elif caseNumber == 9:
                booleanTexturecaseNumber.SetInIn(clearTexture)
                booleanTexturecaseNumber.SetInOut(solidTexture)
                booleanTexturecaseNumber.SetOutIn(solidTexture)
                booleanTexturecaseNumber.SetOutOut(clearTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(edgeTexture)
                booleanTexturecaseNumber.SetOnOut(edgeTexture)
                booleanTexturecaseNumber.SetInOn(edgeTexture)
                booleanTexturecaseNumber.SetOutOn(edgeTexture)
            elif caseNumber == 10:
                booleanTexturecaseNumber.SetInIn(solidTexture)
                booleanTexturecaseNumber.SetInOut(solidTexture)
                booleanTexturecaseNumber.SetOutIn(clearTexture)
                booleanTexturecaseNumber.SetOutOut(clearTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(edgeTexture)
                booleanTexturecaseNumber.SetOnOut(edgeTexture)
                booleanTexturecaseNumber.SetInOn(solidTexture)
                booleanTexturecaseNumber.SetOutOn(clearTexture)
            elif caseNumber == 11:
                booleanTexturecaseNumber.SetInIn(clearTexture)
                booleanTexturecaseNumber.SetInOut(solidTexture)
                booleanTexturecaseNumber.SetOutIn(clearTexture)
                booleanTexturecaseNumber.SetOutOut(clearTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(clearTexture)
                booleanTexturecaseNumber.SetOnOut(edgeTexture)
                booleanTexturecaseNumber.SetInOn(edgeTexture)
                booleanTexturecaseNumber.SetOutOn(clearTexture)
            elif caseNumber == 12:
                booleanTexturecaseNumber.SetInIn(solidTexture)
                booleanTexturecaseNumber.SetInOut(clearTexture)
                booleanTexturecaseNumber.SetOutIn(solidTexture)
                booleanTexturecaseNumber.SetOutOut(clearTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(solidTexture)
                booleanTexturecaseNumber.SetOnOut(clearTexture)
                booleanTexturecaseNumber.SetInOn(edgeTexture)
                booleanTexturecaseNumber.SetOutOn(edgeTexture)
            elif caseNumber == 13:
                booleanTexturecaseNumber.SetInIn(clearTexture)
                booleanTexturecaseNumber.SetInOut(clearTexture)
                booleanTexturecaseNumber.SetOutIn(solidTexture)
                booleanTexturecaseNumber.SetOutOut(clearTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(edgeTexture)
                booleanTexturecaseNumber.SetOnOut(clearTexture)
                booleanTexturecaseNumber.SetInOn(clearTexture)
                booleanTexturecaseNumber.SetOutOn(edgeTexture)
            elif caseNumber == 14:
                booleanTexturecaseNumber.SetInIn(solidTexture)
                booleanTexturecaseNumber.SetInOut(clearTexture)
                booleanTexturecaseNumber.SetOutIn(clearTexture)
                booleanTexturecaseNumber.SetOutOut(clearTexture)
                booleanTexturecaseNumber.SetOnOn(edgeTexture)
                booleanTexturecaseNumber.SetOnIn(edgeTexture)
                booleanTexturecaseNumber.SetOnOut(clearTexture)
                booleanTexturecaseNumber.SetInOn(edgeTexture)
                booleanTexturecaseNumber.SetOutOn(clearTexture)
            elif caseNumber == 15:
                booleanTexturecaseNumber.SetInIn(clearTexture)
                booleanTexturecaseNumber.SetInOut(clearTexture)
                booleanTexturecaseNumber.SetOutIn(clearTexture)
                booleanTexturecaseNumber.SetOutOut(clearTexture)
                booleanTexturecaseNumber.SetOnOn(clearTexture)
                booleanTexturecaseNumber.SetOnIn(clearTexture)
                booleanTexturecaseNumber.SetOnOut(clearTexture)
                booleanTexturecaseNumber.SetInOn(clearTexture)
                booleanTexturecaseNumber.SetOutOn(clearTexture)


            booleanTexturecaseNumber.Update()
            return booleanTexturecaseNumber

        # A list of positions
        positions = []
        positions.append((-4, 4, 0))
        positions.append((-2, 4, 0))
        positions.append((0, 4, 0))
        positions.append((2, 4, 0))
        positions.append((-4, 2, 0))
        positions.append((-2, 2, 0))
        positions.append((0, 2, 0))
        positions.append((2, 2, 0))
        positions.append((-4, 0, 0))
        positions.append((-2, 0, 0))
        positions.append((0, 0, 0))
        positions.append((2, 0, 0))
        positions.append((-4, -2, 0))
        positions.append((-2, -2, 0))
        positions.append((0, -2, 0))
        positions.append((2, -2, 0))

        ren = vtkRenderer()
        renWin = vtkRenderWindow()
        renWin.AddRenderer(ren)

        # define two elliptical cylinders
        quadric1 = vtkQuadric()
        quadric1.SetCoefficients(1, 2, 0, 0, 0, 0, 0, 0, 0, -.07)

        quadric2 = vtkQuadric()
        quadric2.SetCoefficients(2, 1, 0, 0, 0, 0, 0, 0, 0, -.07)

        # create a sphere for all to use
        aSphere = vtkSphereSource()
        aSphere.SetPhiResolution(50)
        aSphere.SetThetaResolution(50)

        # create texture coordinates for all
        tcoords = vtkImplicitTextureCoords()
        tcoords.SetInputConnection(aSphere.GetOutputPort())
        tcoords.SetRFunction(quadric1)
        tcoords.SetSFunction(quadric2)

        aMapper = vtkDataSetMapper()
        aMapper.SetInputConnection(tcoords.GetOutputPort())

        # create a mapper, sphere and texture map for each case
        aTexture = []
        anActor = []
        for i in range(0, 16):
            aTexture.append(vtkTexture())
            aTexture[i].SetInputData(makeBooleanTexture(i, 256, 1).GetOutput())
            aTexture[i].InterpolateOff()
            aTexture[i].RepeatOff()
            anActor.append(vtkActor())
            anActor[i].SetMapper(aMapper)
            anActor[i].SetTexture(aTexture[i])
            anActor[i].SetPosition(positions[i])
            anActor[i].SetScale(2.0, 2.0, 2.0)
            ren.AddActor(anActor[i])


        ren.SetBackground(0.4392, 0.5020, 0.5647)
        ren.ResetCamera()
        ren.GetActiveCamera().Zoom(1.4)
        renWin.SetSize(500, 500)

        # render and interact with data

        iRen = vtkRenderWindowInteractor()
        iRen.SetRenderWindow(renWin);
        renWin.Render()

        img_file = "quadricCut.png"
        vtkmodules.test.Testing.compareImage(iRen.GetRenderWindow(), vtkmodules.test.Testing.getAbsImagePath(img_file), threshold=25)
        vtkmodules.test.Testing.interact()

if __name__ == "__main__":
     vtkmodules.test.Testing.main([(quadricCut, 'test')])
